app/models/user.rb

```diff
@@ -7,17 +7,64 @@ class User < ApplicationRecord
   # end
 
   def starred_repos
-    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    parse(Faraday.get("https://api.github.com/#{self[:username]}/starred?token=#{self.token}"))
   end
 
   def followers
-   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers#{auth}"))
   end
-  
+
   def following
-  parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following#{auth}"))
+  end
+
+  def repos
+    client_id     = ENV["GITHUB_CLIENT_ID"]
```

Comment:
@cmacaulay They are two different options.  Conceptually you are finding the repos that belong to a user. so user.repos makes sense - similar to building a one to many relationship. I think actually if you build it off the user model you don't need it go to through the controller since you could call currrent_user.repos. The flow might look like this: 
Current_user.repos (in view) > calls to Repo PORO > calls to GitHubService > returns back to PORO to turn hash into Repo Objects > returns back User as an array of Repo objects> returns to the view to be displayed. 

Does that help at all?